projects/knora-ui/src/lib/viewer/values/boolean-value/boolean-value.component.ts

```diff
@@ -0,0 +1,125 @@
+import { Component, Inject, Input, OnChanges, OnDestroy, OnInit, SimpleChanges } from '@angular/core';
+import { FormBuilder, FormControl, FormGroup } from '@angular/forms';
+import { MatCheckboxChange } from '@angular/material/typings';
+import { CreateBooleanValue, ReadBooleanValue, UpdateBooleanValue } from '@knora/api';
+import { Subscription } from 'rxjs';
+import { BaseValueComponent } from '../base-value.component';
+
+@Component({
+  selector: 'kui-boolean-value',
+  templateUrl: './boolean-value.component.html',
+  styleUrls: ['./boolean-value.component.scss']
+})
+export class BooleanValueComponent extends BaseValueComponent implements OnInit, OnChanges, OnDestroy {
+
+  @Input() displayValue?: ReadBooleanValue;
+
+  valueFormControl: FormControl;
+  commentFormControl: FormControl;
+
+  form: FormGroup;
+
+  valueChangesSubscription: Subscription;
+
+  customValidators = [];
+
+  booleanLabel: string;
+
+  constructor(@Inject(FormBuilder) private fb: FormBuilder) {
+    super();
+  }
+
+  getInitValue(): boolean | null {
+    if (this.displayValue !== undefined) {
+      return this.displayValue.bool;
+    } else {
+      return null;
+    }
+  }
+
+  ngOnInit() {
+    // initialize form control elements
+    this.valueFormControl = new FormControl(null);
+    this.commentFormControl = new FormControl(null);
+
+    // subscribe to any change on the comment and recheck validity
+    this.valueChangesSubscription = this.commentFormControl.valueChanges.subscribe(
+      data => {
+        this.valueFormControl.updateValueAndValidity();
+      }
+    );
+
+    this.form = this.fb.group({
+      booleanValue: this.valueFormControl,
+      comment: this.commentFormControl
+    });
+
+    this.resetFormControl();
+  }
+
+  // override the resetFormControl() from the base component to deal with the disabled state and the checkbox label
+  resetFormControl(): void {
+    super.resetFormControl();
+
+    if (this.valueFormControl !== undefined) {
+      this.booleanLabel = this.valueFormControl.value ? 'true' : 'false';
```

Comment:
Could you create a method containing `this.booleanLabel = this.valueFormControl.value ? 'true' : 'false';` and call it from here and from `onChecked `?